cluster/ubuntu/util.sh

```diff
@@ -175,6 +177,8 @@ function verify-minion(){
         ssh "$1" "pgrep -f $daemon" >/dev/null 2>&1 || {
         printf "."
         validated="1"
+        # Try to start the service again if it's not running.
+        ssh "$1" "sudo service $daemon start" >/dev/null 2>&1 || true
```

Comment:
I don't think that restart logic belongs to 'verify' function. This should be 'read-only' in a sense. Please move it to the upper layer.